source/lex.tex

```diff
@@ -1396,7 +1405,7 @@
 \begin{bnf}
 \nontermdef{hexadecimal-fractional-constant}\br
     \opt{hexadecimal-digit-sequence} \terminal{.} hexadecimal-digit-sequence\br
-    hexadecimal-digit-sequence \terminal{.}
+    hexadecimal-digit-sequence \opt{\terminal{.}}
```

Comment:
Hmm. I can see why this was done the way it was (to make the decimal and hexadecimal cases more parallel). Do you think this is worthwhile?

Reply:
It's a bit unfortunate that we don't have a good way to talk about the significand in the decimal case, because it's two grammar productions. Other than that, the old grammar just seemed needlessly verbose.  If you feel the axe was too large for that particular part, I can undo the change.

Reply:
I would prefer to keep the grammar productions parallel between the decimal and hexadecimal case. This also keeps our lexical grammar matching C's in this regard, which seems valuable. So yes, please revert the grammar change.

Reply:
Grammar change reverted.

Reply:
I've also augmented the definition of "significand" to account for the two different grammar options.